lib/_http_outgoing.js

```diff
@@ -337,6 +337,14 @@ OutgoingMessage.prototype.getHeader = function(name) {
 };
 
 
+OutgoingMessage.prototype.getAllHeaders = function() {
+  if (!this._headers)
+    return;
+  else
+    return util._extend({}, this._headers);
```

Comment:
@chrisdickinson Of course.

The discussion is weather to extend `{}` or `Object.create(null)`. :)